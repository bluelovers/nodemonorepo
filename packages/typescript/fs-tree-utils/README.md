# fs-tree-utils

Work with filesystem tree with ease

## APIs

**NOTE:**
  * `fsTreeUtils` is an alias of module `fs-tree-utils`.
  * `fsExtra` is an alias of module `fs-extra`.
  * `fs` is an alias of module `fs`.
  * All example code snippets are in async context.

### `fsTreeUtils.traverse`

```typescript
declare function traverse (dirname: string, options?: Options): Promise<DeepFuncParam[]>

interface Options {
  readonly deep?: DeepFunc
  readonly level?: number
  readonly stat?: StatFunc
}

interface DeepFuncParam {
  readonly container: string
  readonly item: string
  readonly path: string
  readonly stats: fsx.Stats
  readonly level: Level
}

type DeepFunc = (param: DeepFuncParam) => boolean
type Stat = (path: string) => fs.Stats | Promise<fs.Stats>

// See https://git.io/vhar7 for more types
```

**Parameters:**
  * `dirname`: Directory that contains all items need to be iterated over.
  * `options.deep` (optional): Decides whether or not to dive deeper, default to "always".
  * `options.level` (optional): Initial level of depth, default to `0`.
  * `options.stat` (optional): Return filesystem item's stats (`fs.Stats`), default to `fs.stat`.

**Returns:**
  * A promise of `DeepFuncParam[]`

**Other types:**
  * `DeepFuncParam::container`: Path to directory that contains each item.
  * `DeepFuncParam::item`: Name (basename) of each item (including extension).
  * `DeepFuncParam::path`: Full path to each item.
  * `DeepFuncParam::stats`: Returned value of `options.stat` upon each item.
  * `DeepFuncParam::level`: Level of depth, minimum depth is customizable via `options.level`.

#### Examples

##### Simplest form

```javascript
import {traverse} from 'fs-tree-utils'
const res = await traverse('/path/to/my/directory')
console.log('all basenames', x => x.item)
console.log('all last changes', x => x.stats.mtime)
console.log('all fullpaths', x => x.path)
```

##### Get all symbolic links

This example requires `fsExtra.lstat`/`fs.lstatSync` in order to detect symbolic links. This is also safer methods when there're recursive links.

```javascript
import {traverse} from 'fs-tree-utils'
import {lstatSync} from 'fs' // you can also use `fsExtra.lstat` for true async operation
const res = await traverse('/path/to/my/directory', {stat: lstatSync})
const links = res.filter(x => x.stats.isSymbolicLink()).map(x => x.path)
console.log('all symlinks', links)
```

##### With specified `deep`

Don't traverse `.git`'s/`node_modules`'s children.

```javascript
import {traverse} from 'fs-tree-utils'
const deep = ({item}) => item !== '.git' && item !== 'node_modules'
const res = await traverse(`/path/to/my/directory`, {deep})
console.log('result', res)
```

### `fsTreeUtils.read.nested`

```typescript
declare function readNested (name: string, options?: NestedReadOptions): Promise<Tree.Read.Node>

interface NestedReadOptions {
  readonly stat?: NestedReadOptions.StatFunc
  readonly onerror?: NestedReadOptions.ErrorHandler
  readonly onunknown?: NestedReadOptions.UnknownHandler
}

// See https://git.io/vhar7 for more types
```

**Parameters:**
  * `name`: Path to the top directory of a tree.
  * `options.stat` (optional): Stat function to use (returns either `fs.Stats` or `Promise<fs.Stats>`), default to `fs.stat`.
  * `options.onerror` (optional): Function that transforms an error into an instance of `fsTreeUtils.Exception.ErrorCarrier`, these errors will be thrown instead if the function is not provided.
  * `options.onunknown` (optional): Function that creates an instance of `fsTreeUtils.Exception.Other` when encounter unknown filesystem entity, errors would be thrown instead if the function is not provided.

#### Examples

##### Simplest form

```javascript
import {read} from 'fs-tree-utils'
const tree = await read.nested('/path/to/my/directory')
console.log('Structure of "/path/to/my/directory"', tree)
```

##### With specified `options.stat`

Use `fsExtra.lstat` as `options.stat`.

```javascript
import {read} from 'fs-tree-utils'
import {lstat} from 'fs-extra' // function lstat (path: string): Promise<fs.Stats>
const tree = await read.nested('/path/to/my/directory', {stat: lstat})
console.log('Structure of "/path/to/my/directory"', tree)
```

### `fsTreeUtils.read.flat`

```typescript
declare function readFlat (name: string, options?: Traverse.Options): Promise<FlatReadResultValue[]>

interface FlatReadResultValue {
  readonly fileContents: FlatReadResultFileContent
  readonly directories: string[]
  readonly files: string[]
  readonly all: string[]
}

interface FlatReadResultFileContent {
  readonly [filename: string]: Tree.Read.FileContent
}

// See https://git.io/vhar7 for more types
```

This function uses `traverse` under the hook.

**Parameters:**
  * `name`: Path to the top directory.
  * `options` (optional): Options to pass to `traverse`.

**Returns:**
  * `FlatReadResultValue::fileContents`: A dict (`Object`) with keys being files' paths (`string`) and values being files' content (`Buffer | string`).
  * `FlatReadResultValue::directories`: A list (`Array`) of directories' paths (`string`).
  * `FlatReadResultValue::files`: A list (`Array`) of files' paths (`string`).
  * `FlatReadResultValue::all`: A list (`Array`) of every item's path (`string`).

### `fsTreeUtils.create`

```typescript
declare function create (tree: Tree.Write, container: string): Promise<void>

// `Tree.Write` is basically tree-like dictionary, see https://git.io/vhar7 for its definition
```

**Parameters:**
  * `tree`: Intended tree structure representation.
  * `container`: Intended top-level directory.

**Returns:**
  * A promise that resolves when tree creation is finishes successfully and rejects when tree creation fails.

**Effects:**
  * Populate directory of `container` with filesystem entities that as represented in `tree`.

#### Examples

##### Tree Creation

```javascript
import {
  create,
  FileSystemRepresentation
} from 'fs-tree-utils'

import {
  writeFile,
  writeFileSync
} from 'fs-extra'

const tree = { // top-level `tree` object corresponds to top-level directory of `container` (directory)
  files: { // corresponding path: {container}/files/ (directory)
    'string.txt': 'A string', // corresponding path: {container}/files/string.txt (file)
    'buffer.txt': Buffer.from('A buffer'), // corresponding path: {container}/files/buffer.txt (file)
    'class.txt': new FileSystemRepresentation.File('File class'), // corresponding path: {container}/files/class.txt (file)
    'function.async.txt': name => writeFile(name), // corresponding path: {container}/files/function.async.txt
    'function.sync.txt': name => writeFileSync(name) // corresponding path: {container}/files/function.sync.txt
  },
  symlinks: { // corresponding path: {container}/symlinks
    'to-files-container': new FileSystemRepresentation.Symlink('../files'), // points to {container}/files/
    'to-nowhere': new FileSystemRepresentation.Symlink('/this/is/nowhere'), // points to /this/is/nowhere
    'a-windows-junction': new FileSystemRepresentation.Symlink('C:\\Windows', {type: 'junction'}) // this is a junction (in Windows)
  },
  clones: { // corresponding path: {container}/clones
    'my-copy': new FileSystemRepresentation.Clone('/source/for/copying') // uses `fsExtra.copy` under the hook
  }
}
```

### `fsTreeUtils.FileSystemRepresentation`

```typescript
abstract class FileSystemRepresentation {
  abstract public write (target: string, param: CreateSecondParam): Promise<void> | void
}

interface CreateSecondParam {
  create (tree: Tree.Write, container: string): Promise<void>
}
```

This is an abstract class upon which `fsTreeUtils.FileSystemRepresentation.*` was built.

#### `fsTreeUtils.FileSystemRepresentation::write`

This is an abstract method.

This method is used by `fsTreeUtils.create`.

**Parameters:**
  * `target`: Name of represented entity.
  * `param.create`: It is `fsTreeUtils.create`.

**Returns:**
  * Synchronous version returns nothing (i.e. `void` a.k.a. `undefined`).
  * Asynchronous version returns a promise that resolves `undefined`.

### `fsTreeUtils.FileSystemRepresentation.File`

```typescript
class File extends FileSystemRepresentation {
  public constructor (content: string | Buffer)
  public write (filename: string): Promise<void>
}
```

#### `fsTreeUtils.FileSystemRepresentation.File::constructor`

**Parameters:**
  * `content`: Intended file's content (`string | Buffer`).

#### `fsTreeUtils.FileSystemRepresentation.File::write`

**Parameters:**
  * `filename`: Intended file's name (`string`).

**Returns:**
  * A promise that resolves when content is written into file successfully.

**Effects:**
  * Write `content` into file with name `filename`.

### Other types and classes

See [module `fs-tree-utils/lib/.types`](https://git.io/vhar7).
