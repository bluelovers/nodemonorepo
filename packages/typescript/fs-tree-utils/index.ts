import traverse from './lib/traverse'
import read from './lib/read'
import create from './lib/create'
export * from './lib/.types'
export { traverse, read, create }
