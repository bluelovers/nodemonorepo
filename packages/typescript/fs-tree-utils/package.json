{
  "name": "fs-tree-utils",
  "version": "0.0.5",
  "description": "Work with filesystem tree with ease",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "filesystem",
    "tree"
  ],
  "main": "index.js",
  "dependencies": {
    "fs-extra": "^6.0.1",
    "ramda": "^0.25.0",
    "@types/fs-extra": "^5.0.2",
    "@types/ramda": "^0.25.28",
    "@types/node": "^10.0.8"
  },
  "devDependencies": {
    "unique-temp-path": "*"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
