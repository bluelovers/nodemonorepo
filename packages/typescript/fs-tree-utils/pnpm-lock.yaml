dependencies:
  '@types/fs-extra': 8.0.0
  '@types/node': 12.7.5
  '@types/ramda': 0.26.21
  fs-extra: 8.1.0
  monorepo-shared-assets: 'link:../monorepo-shared-assets'
  ramda: 0.26.1
  tslib: 1.10.0
devDependencies:
  unique-temp-path: 'link:../unique-temp-path'
lockfileVersion: 5.1
packages:
  /@types/fs-extra/8.0.0:
    dependencies:
      '@types/node': 12.7.5
    dev: false
    resolution:
      integrity: sha512-bCtL5v9zdbQW86yexOlXWTEGvLNqWxMFyi7gQA7Gcthbezr2cPSOb8SkESVKA937QD5cIwOFLDFt0MQoXOEr9Q==
  /@types/node/12.7.5:
    dev: false
    resolution:
      integrity: sha512-9fq4jZVhPNW8r+UYKnxF1e2HkDWOWKM5bC2/7c9wPV835I0aOrVbS/Hw/pWPk2uKrNXQqg9Z959Kz+IYDd5p3w==
  /@types/ramda/0.26.21:
    dependencies:
      ts-toolbelt: 3.8.92
    dev: false
    resolution:
      integrity: sha512-zMYtIZMceA6BvH+or6LmewLBgojbXg5+FGCwjO8K+Z+d/ZWxILmhhASXkehW0PqJL+V0QbyDeeAHix0dvEKXfQ==
  /fs-extra/8.1.0:
    dependencies:
      graceful-fs: 4.2.2
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false
    engines:
      node: '>=6 <7 || >=8'
    resolution:
      integrity: sha512-yhlQgA6mnOJUKOsRUFsgJdQCvkKhcz8tlZG5HBQfReYZy46OwLcY+Zia0mtdHsOo9y/hP+CxMN0TU9QxoOtG4g==
  /graceful-fs/4.2.2:
    dev: false
    resolution:
      integrity: sha512-IItsdsea19BoLC7ELy13q1iJFNmd7ofZH5+X/pJr90/nRoPEX0DJo1dHDbgtYWOhJhcCgMDTOw84RZ72q6lB+Q==
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.2.2
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /ramda/0.26.1:
    dev: false
    resolution:
      integrity: sha512-hLWjpy7EnsDBb0p+Z3B7rPi3GDeRG5ZtiI33kJhTt+ORCd38AbAIjB/9zRIUoeTbE/AVX5ZkU7m6bznsvrf8eQ==
  /ts-toolbelt/3.8.92:
    dev: false
    resolution:
      integrity: sha512-8weyhjSrpxBdkcSZF4qV6aq/q7S795dKS3SWxGhAs7msilb3eVN8ozuXBlDR5ivQVFvp9cfgwXb+Nb4aKEg0tg==
  /tslib/1.10.0:
    dev: false
    resolution:
      integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==
  /universalify/0.1.2:
    dev: false
    engines:
      node: '>= 4.0.0'
    resolution:
      integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==
specifiers:
  '@types/fs-extra': ^8.0.0
  '@types/node': ^12.7.5
  '@types/ramda': ^0.26.21
  fs-extra: ^8.1.0
  monorepo-shared-assets: ^0.0.17
  ramda: ^0.26.1
  tslib: ^1.10.0
  unique-temp-path: '*'
