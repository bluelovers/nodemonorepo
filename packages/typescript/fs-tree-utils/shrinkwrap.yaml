dependencies:
  '@types/fs-extra': 5.0.1
  '@types/node': 9.6.4
  '@types/ramda': 0.25.21
  extra-jest: 'link:../extra-jest'
  fs-extra: 5.0.0
  ramda: 0.25.0
  unique-temp-path: 'link:../unique-temp-path'
packages:
  /@types/fs-extra/5.0.1:
    dependencies:
      '@types/node': 9.6.2
    dev: false
    resolution:
      integrity: sha512-h3wnflb+jMTipvbbZnClgA2BexrT4w0GcfoCz5qyxd0IRsbqhLSyesM6mqZTAnhbVmhyTm5tuxfRu9R+8l+lGw==
  /@types/node/9.6.2:
    dev: false
    resolution:
      integrity: sha512-UWkRY9X7RQHp5OhhRIIka58/gVVycL1zHZu0OTsT5LI86ABaMOSbUjAl+b0FeDhQcxclrkyft3kW5QWdMRs8wQ==
  /@types/node/9.6.4:
    dev: false
    resolution:
      integrity: sha512-Awg4BcUYiZtNKoveGOu654JVPt11V/KIC77iBz8NweyoOAZpz5rUJfPDwwD+ajfTs2HndbTCEB8IuLfX9m/mmw==
  /@types/ramda/0.25.21:
    dev: false
    resolution:
      integrity: sha512-y4Dx3t+EA5Y+s87cwUoIzo0Tsj9z7QCu4gaXrdFln4wQRgk9igBskEEe7gZ0PG8dSXtQb29oA3RS//3XquHdiw==
  /fs-extra/5.0.0:
    dependencies:
      graceful-fs: 4.1.11
      jsonfile: 4.0.0
      universalify: 0.1.1
    dev: false
    resolution:
      integrity: sha512-66Pm4RYbjzdyeuqudYqhFiNBbCIuI9kgRqLPSHIlXHidW8NIQtVdkM1yeZ4lXwuhbTETv3EUGMNHAAw6hiundQ==
  /graceful-fs/4.1.11:
    dev: false
    engines:
      node: '>=0.4.0'
    resolution:
      integrity: sha1-Dovf5NHduIVNZOBOp8AOKgJuVlg=
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.1.11
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /ramda/0.25.0:
    dev: false
    resolution:
      integrity: sha512-GXpfrYVPwx3K7RQ6aYT8KPS8XViSXUVJT1ONhoKPE9VAleW42YE+U+8VEyGWt41EnEQW7gwecYJriTI0pKoecQ==
  /universalify/0.1.1:
    dev: false
    resolution:
      integrity: sha1-+nG63UQ3r0wUiEHjs7Fl+enlkLc=
registry: 'https://registry.npmjs.org/'
shrinkwrapMinorVersion: 5
shrinkwrapVersion: 3
specifiers:
  '@types/fs-extra': ^5.0.1
  '@types/node': ^9.6.4
  '@types/ramda': ^0.25.21
  extra-jest: '*'
  fs-extra: ^5.0.0
  ramda: ^0.25.0
  unique-temp-path: '*'
