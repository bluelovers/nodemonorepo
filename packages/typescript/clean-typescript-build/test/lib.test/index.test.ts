import * as fsTreeUtils from 'fs-tree-utils'
import {clean, listAllTargets} from '../../index'
import createVirtualEnv from '../.lib/virtual-env'

const {apply} = createVirtualEnv()

it('should list correct target files', apply(async () => {
  const [all, targets] = await Promise.all([
    fsTreeUtils.read.flat('root'),
    listAllTargets('root')
  ])

  const deletedFileContents = targets.map(filename => ({
    filename,
    content: all.fileContents[filename]
  }))

  const remainingFileContents = all.files
    .filter(filename => !targets.includes(filename) && !/__describe$/.test(filename))
    .map(filename => ({filename, content: all.fileContents[filename]}))

  expect({
    targets,
    deletedFileContents,
    remainingFileContents
  }).toMatchSnapshot()

  expect({
    invalidAcceptance: remainingFileContents.filter(x => x.content !== 'Should be kept'),
    invalidDeletion: deletedFileContents.filter(x => x.content !== 'Should be deleted')
  }).toEqual({
    invalidAcceptance: [],
    invalidDeletion: []
  })
}))
