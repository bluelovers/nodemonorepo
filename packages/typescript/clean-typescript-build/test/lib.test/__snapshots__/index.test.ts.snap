// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should list correct target files 1`] = `
Object {
  "deletedFileContents": Array [
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/incompleted-out/file.js",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.js",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.js.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.jsx",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.jsx.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.d.ts",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.d.ts.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.d.tsx",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.d.tsx.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.js",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.js.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.jsx",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.jsx.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.d.ts",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.d.ts.map",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.d.tsx",
    },
    Object {
      "content": "Should be deleted",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.d.tsx.map",
    },
  ],
  "remainingFileContents": Array [
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-cleaned/src-and-out/incompleted-out/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-cleaned/src-and-out/ts-ext/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-cleaned/src-and-out/tsx-ext/file.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/folders/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.d.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.d.ts.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.d.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.d.tsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.js",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.js.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.jsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.jsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/direct-node-modules-descendants/node_modules/file.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.d.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.d.ts.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.d.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.d.tsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.js",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.js.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.jsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.jsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/ignored/indirect-node-modules-descendants/node_modules/a-dummy-package/file.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.d.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.d.ts.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.d.tsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.d.tsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.js",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.js.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.jsx",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/out-only/file.jsx.map",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/src-only/file.ts",
    },
    Object {
      "content": "Should be kept",
      "filename": "root/to-be-kept/src-only/file.tsx",
    },
  ],
  "targets": Array [
    "root/to-be-cleaned/src-and-out/incompleted-out/file.js",
    "root/to-be-cleaned/src-and-out/ts-ext/file.js",
    "root/to-be-cleaned/src-and-out/ts-ext/file.js.map",
    "root/to-be-cleaned/src-and-out/ts-ext/file.jsx",
    "root/to-be-cleaned/src-and-out/ts-ext/file.jsx.map",
    "root/to-be-cleaned/src-and-out/ts-ext/file.d.ts",
    "root/to-be-cleaned/src-and-out/ts-ext/file.d.ts.map",
    "root/to-be-cleaned/src-and-out/ts-ext/file.d.tsx",
    "root/to-be-cleaned/src-and-out/ts-ext/file.d.tsx.map",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.js",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.js.map",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.jsx",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.jsx.map",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.d.ts",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.d.ts.map",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.d.tsx",
    "root/to-be-cleaned/src-and-out/tsx-ext/file.d.tsx.map",
  ],
}
`;
